src/System.Management.Automation/engine/CommandProcessorBase.cs

```diff
@@ -169,24 +169,38 @@ internal bool UseLocalScope
             InvocationInfo invocationInfo)
         {
             // If we are in a constrained language mode (Core or Restricted), block it.
-            // This goes both ways:
+            // We are currently restricting in one direction:
             //    - Can't dot something from a more permissive mode, since that would probably expose
             //      functions that were never designed to handle untrusted data.
-            //    - Can't dot something from a less permissive mode, since that might introduce tainted
-            //      data into the current scope.
```

Comment:
This comment is a little confusing. By looking at the code, the script block may from a more permissive mode (FullLanguage) or a less permissive mode (NoLanguage), but as long as it's just powershell commands plus parameters then it's allowed to run. So it seems there is really no one direction restriction.